src/realm/sync/instruction_replication.cpp

```diff
@@ -750,7 +761,22 @@ void SyncReplication::populate_path_instr(Instruction::PathInstruction& instr, c
 {
     REALM_ASSERT(key);
     // The first path entry will be the column key
-    REALM_ASSERT(path[0].is_col_key());
+    std::string field_name;
+    if (path[0].is_col_key()) {
+        auto ck = path[0].get_col_key();
+        if (table.is_additional_props_col(ck)) {
+            REALM_ASSERT(path.size() > 1);
+            field_name = path[1].get_key();
+            // Erase the "__additional__" part of the path
+            path.erase(path.begin());
+        }
+        else {
+            field_name = table.get_column_name(ck);
+        }
+    }
+    else {
+        field_name = path[0].get_key();
```

Comment:
when can this happen?

Reply:
In the case of an additional property directly on an object, the first element is a string.

Reply:
I am not sure I understand. Isn't the additional column always involved when we're dealing with additional properties?

Reply:
Yes, but that is handled in `SyncReplication::dictionary_update`

Reply:
I was about to write that because of the change in `dictionary_update`, the path does not always start with a ColKey anymore.